src/shared/components/layouts/StickToTop.tsx

```diff
@@ -44,56 +46,33 @@ export const StickToTop = ({
       setWrapperElement(wrapperElement)
     }
 
-    if (scrollContainerId) {
-      const element = document.getElementById(scrollContainerId)
-      if (element) {
-        element.addEventListener('scroll', handleScroll)
-      }
-
-      return () => {
-        if (element) {
-          element.removeEventListener('scroll', handleScroll)
-        }
-      }
-    }
-
-    document.addEventListener('scroll', handleScroll)
+    const scrollElement = scrollContainerId ? document.getElementById(scrollContainerId) : window
+    scrollElement?.addEventListener('scroll', handleScroll)
 
     return () => {
-      document.removeEventListener('scroll', handleScroll)
+      scrollElement?.removeEventListener('scroll', handleScroll)
     }
-  }, [scrollContainerId, disable])
+  }, [scrollContainerId, disable, id, palceholderId])
 
-  async function handleScroll(this: HTMLElement) {
-    let scrollTop = 0
+  const handleScroll = useRef(
+    throttle(function (this: HTMLElement) {
+      const currentElement = document.getElementById(stick.current ? palceholderId : id)
 
-    if (scrollContainerId) {
-      scrollTop = this.scrollTop
-    } else {
-      scrollTop = document.scrollingElement?.scrollTop || 0
-    }
-
-    let currentElement
-
-    if (stick.current) {
-      currentElement = document.getElementById(palceholderId)
-    } else {
-      currentElement = document.getElementById(id)
-    }
+      if (currentElement) {
+        const rect = currentElement.getBoundingClientRect()
+        const scrollValue = rect.top - offset
 
-    if (currentElement) {
-      const scrollValue = currentElement.offsetTop - currentElement.scrollTop + currentElement.clientTop - scrollTop
-      if (scrollValue > 0 && scrollValue <= offset + bias) {
-        setStick(true)
-      } else if (scrollValue > offset + bias + buffer) {
-        setStick(false)
+        if (scrollValue <= bias) {
+          setStick(true)
+        } else if (scrollValue > bias + buffer) {
+          setStick(false)
+        }
       }
-    }
-  }
+    }, 16),
+  ).current // 60fps throttle
```

Comment:
Good job here, I think you fixed the logic.

 I tested it out it worked both with and without the throttle function, it worked without issues for both.
 
 There's benefits to using a throttle function, but I don't think it's necessary in this case, with but I leave it up to you.